test/comparator/test_field_comparator.py

```diff
@@ -113,25 +89,57 @@ def test_field_behavior_change(self):
         )
         self.assertEqual(finding.category.name, "FIELD_BEHAVIOR_CHANGE")
 
-    def test_name_change(self):
-        field_foo = make_field("Foo")
-        field_bar = make_field("Bar")
-        FieldComparator(field_foo, field_bar, self.finding_container).compare()
+    def test_primitive_type_change(self):
+        field_int = make_field(proto_type="TYPE_INT32")
+        field_string = make_field(proto_type="TYPE_STRING")
+        FieldComparator(field_int, field_string, self.finding_container).compare()
         finding = self.finding_container.getAllFindings()[0]
         self.assertEqual(
             finding.message,
-            "Name of an existing field is changed from `Foo` to `Bar`.",
+            "Type of an existing field `my_field` is changed from `TYPE_INT32` to `TYPE_STRING`.",
         )
-        self.assertEqual(finding.category.name, "FIELD_NAME_CHANGE")
+        self.assertEqual(finding.category.name, "FIELD_TYPE_CHANGE")
+
+    def test_message_type_change(self):
+        field_message = make_field(type_name=".example.v1.Enum")
+        field_message_update = make_field(type_name=".example.v1beta1.EnumUpdate")
+        FieldComparator(
+            field_message, field_message_update, self.finding_container
+        ).compare()
+        finding = self.finding_container.getAllFindings()[0]
+        self.assertEqual(
+            finding.message,
+            "Type of an existing field `my_field` is changed from `.example.v1.Enum` to `.example.v1beta1.EnumUpdate`.",
+        )
+        self.assertEqual(finding.category.name, "FIELD_TYPE_CHANGE")
+
+    def test_message_type_change_minor_version_update(self):
+        field_message = make_field(type_name=".example.v1.Enum", api_version="v1")
+        field_message_update = make_field(
+            type_name=".example.v1beta1.Enum", api_version="v1beta1"
+        )
+        FieldComparator(
+            field_message, field_message_update, self.finding_container
+        ).compare()
+        findings = self.finding_container.getAllFindings()
+        self.assertFalse(findings)
```

Comment:
Only added one new unit tests for existing field moving into `oneof`. Other changes in `test_field_compatator` is reordering the tests.